### ISSUE TITLE
Add "setting records" to state devices

### ISSUE THREAD

**Issue by rneswold** [fermi-ad/fe-drv-acsys-states#3]
**Is your feature request related to a problem? Please describe.**

The ACSys state front-end doesn't use SQL to determine the devices it owns. Instead, it builds its list of devices as new states are reported. This works well, overall, because this front-end is rarely restarted.

The problem with this approach is there's a window, after the front-end is restarted and before a device is updated, that applications can't read the device (they get a NOT_UPDATED type of status) Depending on how often a particular state device is updated, this window could be rather large.

**Describe the solution you'd like**

If we ensure each state device has a setting record, then all its devices will get set to their previous value soon after the restart. There is still a small window, but transient errors are expected as a front-end comes up.

There's the question whether these initial settings should be multicast as state changes:

- Since the've already been reported, they're just duplicates and would cause client unnecessary work.
- It would cause an unnecessary burst of network traffic.
- HOWEVER, there's a small window where an application sets a state device before the setting record is downloaded. So if we mute the first setting, the application's new state is suppressed and the setting record gets multicast. We can't shut down the SETS32/SETDAT handles until after the download because it uses those handles to send the setting records. Maybe the FSMSET handle can be disabled until the download is complete? That might prevent 99% of the race conditions (which would already be rare.)

**Comment by rneswold** [comment]
I can create a DABBEL file to add the records. @nicklausdj, if I run the script, will the recent DABBEL changes automatically add these devices to the front-end's download list?

**Comment by nicklausdj** [comment]
Yes, it should add them to the download list. You might want to hold off on doing that for a couple days as we’re finding some dabbel problems when deleting the setting records today (John Devoy is looking at it).

From: Rich Neswold ***@***.***>
Date: Tuesday, August 29, 2023 at 2:53 PM
To: fermi-ad/fe-drv-acsys-states ***@***.***>
Cc: Dennis J. Nicklaus ***@***.***>, Mention ***@***.***>
Subject: Re: [fermi-ad/fe-drv-acsys-states] Add "setting records" to state devices (Issue #3)

I can create a DABBEL file to add the records. @nicklausdj<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_nicklausdj&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=O9tjsFcstXfyDppuolvtgs93XYu-awx62AJJHZuGtEM&m=hWwOTTM1rmCfboIcY78QOcxUfMwd6q96iZMw7KjzNQcPE434jsGz8lGh_-gWe2bL&s=YkBCocBkuw4bOZcL0Uo3glbk3X_X4E_ccb2zeCVDEk4&e=>, if I run the script, will the recent DABBEL changes automatically add these devices to the front-end's download list?
